packages/mail-client/src/socket-proxy.ts

```diff
@@ -5,11 +5,26 @@
  * @internal
  */
 // eslint-disable-next-line @typescript-eslint/require-await
-export async function customAuthRequestHandler(jwt: string): Promise<Buffer> {
+export async function customAuthRequestHandler(
+  jwt: string,
+  cloudConnectorLocationId?: string
+): Promise<Buffer> {
   const jwtBytes = [...Buffer.from(jwt)];
   const jwtBytesLength = jwtBytes.length;
+
   const jwtLengthBuffer = Buffer.alloc(4);
   jwtLengthBuffer.writeUInt32BE(jwtBytesLength, 0);
+
+  const cloudConnectorLocationIdBytes = cloudConnectorLocationId
+    ? [...Buffer.from(cloudConnectorLocationId)]
+    : [];
+
+  const cloudConnectorLocationIdLengthBuffer = Buffer.alloc(1);
```

Comment:
[q] Is it intended that this is different than the `jwtLengthBuffer` - I would expect the same format is needed and that the functionality can be extracted into a separate function then. Something with a signature similar to this.

```ts
function createBuffer(): { buffer; length; }
```